.travis.yml

```diff
@@ -1,27 +1,59 @@
 language: rust
-branches:
-  only:
-    - master
-os:
-  - linux
-  - windows
-rust:
-  - stable
-  - beta
-  - nightly
-# Need to cache the whole `.cargo` directory to keep .crates.toml for
-# cargo-update to work, but don't cache the cargo registry
-before_cache:
-  - rm -rf /home/travis/.cargo/registry
+sudo: false
+
 cache:
   directories:
     - /home/travis/.cargo
-before_script:
-  - if [[ "$TRAVIS_RUST_VERSION" == "stable" || "$TRAVIS_RUST_VERSION" == "nightly" ]]; then rustup component add clippy-preview; fi
-script:
-  - if [[ "$TRAVIS_RUST_VERSION" == "stable" || "$TRAVIS_RUST_VERSION" == "nightly" ]]; then cargo clippy --all-targets --all-features -- -D warnings; fi
-  - cargo test --verbose
+
+branches:
+  only:
+    - master
+
+GH_TOKEN: &GH_TOKEN
+  secure:
+
+DEPLOY_TO_GITHUB: &DEPLOY_TO_GITHUB
+  before_deploy:
+    - git config --local user.name "Stuart Crust"
+    - git config --local user.email "stuart.crust@gmail.com"
+    - name="supernova-$TRAVIS_TAG-$TARGET"
+    - mkdir $name
+    - cp target/$TARGET/release/supernova $name/
+    - cp README.md LICENSE-MIT LICENSE-APACHE supernova.sha1 $name/
+    - tar czvf $name.tar.gz $name
+  deploy:
+    provider: releases
+    api_key: *GH_TOKEN
+    file: supernova-$TRAVIS_TAG-$TARGET.tar.gz
+    skip_cleanup: true
+    on:
+      branch: master
+      tags: true
+
 matrix:
-  allow_failures:
-    - rust: nightly
   fast_finish: true
+  include:
+    - name: Linux
+      env: TARGET=x86_64-unknown-linux-gnu
+      rust: stable
+      before_script:
+        - if [[ "$TRAVIS_RUST_VERSION" == "stable" || "$TRAVIS_RUST_VERSION" == "nightly" ]]; then rustup component add clippy-preview; fi
+      script:
+        - if [[ "$TRAVIS_RUST_VERSION" == "stable" || "$TRAVIS_RUST_VERSION" == "nightly" ]]; then cargo clippy --all-targets --all-features -- -D warnings; fi
+        - cargo test --verbose
+        - cargo build --release --target $TARGET --locked
+        - sha1sum target/$TARGET/release/supernova > supernova.sha1
+      <<: *DEPLOY_TO_GITHUB
+      
+    - name: Windows
```

Comment:
https://github.com/danreeves/cargo-cmd/blob/40a032a13ebbffa54de4061713f5a8da99d0537e/.travis.yml, https://github.com/anyshortcut/anyshortcut-cli/blob/78e182cef5ce13c0b8fbfcf21cc2662b3aeadd0b/.travis.yml and https://github.com/jeehoonkang/membarrier-rs/blob/master/.travis.yml used for help with Windows